src/proof/dimacs.cpp

```diff
@@ -97,17 +98,17 @@ std::vector<prop::SatClause> parseDimacs(std::istream& in)
   Assert(nClauses >= 0);
 
   std::vector<prop::SatClause> cnf;
-  for (int i = 0; i < nClauses; ++i)
+  for (uint64_t i = 0; i < nClauses; ++i)
   {
     cnf.emplace_back();
-    int lit;
+    int64_t lit;
     in >> lit;
     Assert(in.good());
     while (lit != 0)
     {
       cnf.back().emplace_back(std::abs(lit) - 1, lit < 0);
       in >> lit;
-      Assert(std::abs(lit) <= nVars);
+      Assert(uint64_t(std::abs(lit)) <= nVars);
```

Comment:
For conversions of primitive types, use `static_cast<X>`.

Reply:
Good catch!